bundles/specmate-migration/src/com/specmate/migration/h2/AttributeToSQLMapper.java

```diff
@@ -0,0 +1,126 @@
+		String alterString = "ALTER TABLE " + objectName + 
+				" ADD COLUMN " + attributeName + 
+				" INTEGER";
+		
+		if (defaultValue != null) {
+			alterString += " DEFAULT " + defaultValue.intValue();
+		}
+		
+		alterDB(alterString, objectName, attributeName, defaultValue);
+	}
+	
+	public void migrateNewDoubleAttribute(String objectName, String attributeName, Double defaultValue) throws SpecmateException {
+		String alterString = "ALTER TABLE " + objectName + 
+				" ADD COLUMN " + attributeName + 
+				" DOUBLE";
+
+		if (defaultValue != null) {
+			alterString += " DEFAULT " + defaultValue;
+		}
+		
+		alterDB(alterString, objectName, attributeName, defaultValue);
+	}
+	
+	public void migrateNewLongAttribute(String objectName, String attributeName, Long defaultValue) throws SpecmateException {
+		String alterString = "ALTER TABLE " + objectName + 
+				" ADD COLUMN " + attributeName + 
+				" BIGINT";
+
+		if (defaultValue != null) {
+			alterString += " DEFAULT " + defaultValue;
+		}
+		
+		alterDB(alterString, objectName, attributeName, defaultValue);
+	}
+	
+	public void migrateNewReference(String objectName, String attributeName) throws SpecmateException {
+		String failmsg = "Migration: Could not add column " + attributeName + " to table " + objectName + ".";
+		String tableNameList = objectName + "_" + attributeName + "_LIST";
+		List<String> queries = new ArrayList<>();
+		queries.add("ALTER TABLE " + objectName + 
+				" ADD COLUMN " + attributeName +
+				" INTEGER");
+		
+		queries.add("CREATE TABLE " + tableNameList + " (" +
+				"CDO_SOURCE BIGINT NOT NULL, " +
+				"CDO_VERSION INTEGER NOT NULL, " +
+				"CDO_IDX INTEGER NOT NULL, " +
+				"CDO_VALUE BIGINT)");
+		
+		queries.add("CREATE UNIQUE INDEX " + 
+				SQLUtil.createRandomIdentifier("PRIMARY_KEY_" + tableNameList) + 
+				" ON " + tableNameList + " (CDO_SOURCE ASC, CDO_VERSION ASC, CDO_IDX ASC)");
+		
+		queries.add("ALTER TABLE " + tableNameList + " ADD CONSTRAINT " + 
+				SQLUtil.createRandomIdentifier("CONSTRAINT_" + tableNameList) + 
+				" PRIMARY KEY (CDO_SOURCE, CDO_VERSION, CDO_IDX)");
+		
+		SQLUtil.executeStatements(queries, connection, failmsg);
+	}
+	
+	public void migrateRenameAttribute(String objectName, String oldAttributeName, String newAttributeName) throws SpecmateException {
+		String failmsg = "Migration: Could not rename column " + oldAttributeName + " in table " + objectName + ".";
+		List<String> queries = new ArrayList<>();
+		queries.add("ALTER TABLE " + objectName + " ALTER COLUMN " + oldAttributeName + " RENAME TO " + newAttributeName);
+		queries.add(renameExternalReference(objectName, oldAttributeName, newAttributeName));
+		SQLUtil.executeStatements(queries, connection, failmsg);
+	}
+	
+	private void alterDB(String alterString, String objectName, String attributeName, Object defaultValue) throws SpecmateException {
```

Comment:
Should you rename this method in something like "setDefaultAndExecute"?

Reply:
Done.